### PR TITLE
feat: Add support for aws secret manager

### LINKED ISSUE #2204: [➕ Feature]: add support for AWS secret manager

### PR DESCRIPTION
**Pull request by HarshitVashisht11**
Closes #2204

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

**Review by talboren — CHANGES_REQUESTED** [review]
Thank you for this! Can you please update https://github.com/keephq/keep/blob/446141d18bf436f370912a03037b253123acdf91/keep/secretmanager/secretmanagerfactory.py? I'll test that this works over AWS.

File: keep/secretmanager/awssecretmanager.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                                "error": str(create_error),
+                            },
+                        )
+                        raise create_error
+                else:
+                    self.logger.error(
+                        "Error writing secret",
+                        extra={"secret_name": secret_name, "error": str(e)},
+                    )
+                    raise e
```
Comment (shahargl) [inline keep/secretmanager/awssecretmanager.py:69]: except also a broad exception (not only client error) and add log

**Comment by shahargl** [comment]
hey! thanks for the contribution.

@HarshitVashisht11 please update docs - https://docs.keephq.dev/deployment/secret-manager

**Review by talboren — APPROVED** [review]
LGTM!

**Comment by HarshitVashisht11** [comment]
Thank you , can you please add a `hacktoberfest-accepted` label

**Pull request merged**